sql-plugin/src/main/spark320/scala/org/apache/spark/sql/rapids/shims/datetimeExpressions.scala

```diff
@@ -133,10 +164,56 @@ case class GpuTimeAdd(start: Expression,
     }
   }
 
+  // A tricky way to check overflow. The result is overflow when positive + positive = negative
+  // or negative + negative = positive, so we can check the sign of the result is the same as
+  // the sign of the operands.
   private def timestampAddDuration(cv: ColumnView, duration: BinaryOperable): ColumnVector = {
```

Comment:
Dulicated.
Could we extract this function into a file like: `datetimeExpressionsUtils.scala`. It seems that it applys for all Spark versions, so do not put this funciton into a shim.

Reply:
Already has overflow check utility:  `AddOverflowChecks.basicOpOverflowCheck`